Sources/CryptorRSA/CryptorRSA.swift

```diff
@@ -723,28 +751,12 @@ public class CryptorRSA {
 					EVP_MD_CTX_free_wrapper(md_ctx)
                 }
                 
-                // convert RSA key to EVP
-                let evp_key = EVP_PKEY_new()
-				var rc = EVP_PKEY_set1_RSA(evp_key, .make(optional: key.reference))
-                guard rc == 1 else {
-                    let source = "Couldn't create key reference from key data"
-                    if let reason = CryptorRSA.getLastError(source: source) {
-                        
-                        throw Error(code: ERR_ADD_KEY, reason: reason)
-                    }
-                    throw Error(code: ERR_ADD_KEY, reason: source + ": No OpenSSL error reported.")
-                }
                 
-                let (md, padding) = algorithm.algorithmForSignature
+                let (md, _) = algorithm.algorithmForSignature
                 
                 // Provide a pkey_ctx to EVP_DigestSignInit so that the EVP_PKEY_CTX of the signing operation
                 // is written to it, to allow alternative signing options to be set
-                var pkey_ctx = EVP_PKEY_CTX_new(evp_key, nil)
-                
-                EVP_DigestSignInit(md_ctx, &pkey_ctx, .make(optional: md), nil, evp_key)
-                
-                // Now that Init has initialized pkey_ctx, set the padding option
-                EVP_PKEY_CTX_ctrl(pkey_ctx, EVP_PKEY_RSA, -1, EVP_PKEY_CTRL_RSA_PADDING, padding, nil)
+                EVP_DigestSignInit(md_ctx, nil, .make(optional: md), nil, .make(optional: key.reference))
```

Comment:
So, as a note to other readers, we discovered through experimentation that the attempt to set the padding algorithm (that is being removed here) 1. fails to actually alter the padding used, and 2. causes the memory allocated by other openssl routines to be leaked.

The default padding algorithm is already `RSA_PKCS1_PADDING`, so there's no need to set it.  Ideally we would be explicit about it (in case the default changes in the future) but for now, this achieves the same as the existing code and does not leak.